src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -25,6 +26,31 @@ public HostString(string value)
         }
 
         /// <summary>
+        /// Creates a new HostString without modification.
+        /// </summary>
+        /// <param name="value">The value should be Unicode rather than punycode. IPv6 addresses must use square braces.</param>
+        /// <param name="port">A positive, greater than 0 value representing the port in the host string.</param>
+        public HostString(string value, int? port)
```

Comment:
Don't use a nullable. If there's no port then make them call the other constructor.